src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -115,7 +117,11 @@ internal ZipArchiveEntry(ZipArchive archive, string entryName)
 
             _compressedSize = 0; // we don't know these yet
             _uncompressedSize = 0;
-            _externalFileAttr = 0;
+            var defaultEntryPermissions = entryName.EndsWith(Path.DirectorySeparatorChar) || entryName.EndsWith(Path.AltDirectorySeparatorChar)
```

Comment:
@eerhardt if you're adding stuff from file system, you go through a function that replaces `Path.DirectorySeparatorChar` and `Path.AltDirectorySeparatorChar` with `/`:

https://github.com/dotnet/runtime/blob/0f75a0269ce14d5838590cb4c634ed685c044d62/src/libraries/Common/src/System/IO/Archiving.Utils.cs#L41-L49

On Linux, `\` is a filename char, and we're replacing it there. Maybe we shouldn't?

For entries created using `ZipArchive.CreateEntry` we're not performing any substitutions.
Maybe we assume the user of this API uses proper separators?
Should we limit to the `Path.DirectorySeparatorChar` for this check?